specifications/wallet/spec-light-mode.md

```diff
@@ -0,0 +1,143 @@
+
+This procedure is implemented and demonstrated in the `light-mode-test` prototype.
+
+## Functionality
+
+### Command line
+
+The `cardano-wallet` executable will feature a new flag `--light` which indicates that the executable is to be started in light-mode:
+
+```
+$ cardano-wallet serve --light CRED
+```
+
+* `CRED` specifies how to connect to the less trusted blockchain data source. (MVP: Use [Blockfrost][]; `CRED` is a filepath to the secret token.)
+* The `--light` argument and the `--node-socket` arguments are mutally exclusive with each other.
+
+### REST API
+
+When the executable was started in light-mode:
+
+* The endpoints in the hierarchy `/v2/byron-wallets/*` MUST return an error. Because byron wallets use random derivation indices to generate addresses, they are significantly more challenging to make a light wallet for.
+* (MVP: The endpoints in the hierarchy `/v2/shared-wallets/*` MAY return an error.)
+* The `GET /v2/network/information` endpoint MAY not return `sync_progress` as a percentage quantity from [0..100].
+
+## Internal
+
+See also the `light-mode-test` prototype!
+
+* Collect required queries on data source in a data type `LightLayer`. In this way, we can replace the data source with a mock data source and run property tests on the chain following logic. (MVP: Use [Blockfrost][] for demonstration purposes.)
+* Provide second implementation of `NetworkLayer` using a `LightLayer`.
+    * See prototype for details! Important changes:
+    * `watchTip` requires polling (2 seconds interval)
+    * `currentNodeEra` may require hard-coding known eras.
+    * `performanceEstimate` requires copying from ledger code.
+    * `syncProgress` should be ignored for MVP.
+    * The node-mode `NetworkLayer` does a lot of caching, we should avoid that for now and instead add a more uniform caching system later  through a function `addCaches :: NetworkLayer -> IO NetworkLayer`.
+* New data type
+
+    ```haskell
+    data BlockSummary m = BlockSummary
+        { from  :: ChainPoint
+        , to    :: ChainPoint
+        , query :: Address -> m [Transaction]
+        }
+    ```
+
+    * consumed by `applyBlocks`. Adapt `discoverTransactions` from the prototype.
+    * produced by `NetworkLayer`. Adapt `lightSync` from the prototype.
+    * Idea: Use an additional GADT wrapper to specialize `applyBlocks` to sequential address states:
+    
+        ```haskell
+            data BlockData m s where
+                List
+                    :: NonEmpty Block
+                    -> BlockData m s
+                Summary
+                    :: BlockSummary m
+                    -> BlockData m (SeqState n k)
+        ```
+        
+        By using this type as argument to `applyBlocks`, we are guaranteed that `BlockSummary` is only used in conjunction with sequential state. Caveat: It would be better if we could avoid parametrizing the `NetworkLayer` type with the address discovery state `s`.
+
+
```

Comment:
One more thing that we should mention.
Rather than using LocalTxSubmission on cardano-node, we will be asking a web service to submit our transactions.
An example of such a thing is `cardano-submit-api`.
We assume this web service will be operated and paid for by someone.